File: src/main/java/gargoyle/ct/prop/impl/ro/CTReadOnlyEnumProperty.java
package gargoyle.ct.prop.impl.ro;

import org.jetbrains.annotations.NotNull;

public class CTReadOnlyEnumProperty<E extends Enum<E>> extends CTBaseReadOnlyProperty<E> {
    public CTReadOnlyEnumProperty(Class<E> type, @NotNull E value) {
        super(type.getSimpleName(), value);
    }

    public CTReadOnlyEnumProperty(String name, @NotNull E value) {
        super(name, value);
    }

    public CTReadOnlyEnumProperty(@NotNull E value) {
        super(value.getClass().getSimpleName(), value);
    }
}

File: src/main/kotlin/gargoyle/ct/prop/impl/ro/CTReadOnlyEnumProperty.kt
package gargoyle.ct.prop.impl.ro

import kotlin.reflect.KClass

class CTReadOnlyEnumProperty<E : Enum<E>> : CTBaseReadOnlyProperty<E> {
    constructor(type: KClass<E>, value: E) : super(type.qualifiedName!!.substringAfterLast('.'), value)
    constructor(name: String, value: E) : super(name, value)
    constructor(value: E) : super(value.javaClass.simpleName, value)
}
